tests/Unit/User/Infrastructure/Factory/RecoveryCodeBatchFactoryTest.php

```
@@ -128,4 +176,42 @@ public function testCreateReturnsUniqueCodesWithHighProbability(): void
+        );
+    }
+
+    private function queueRandomBytes(string ...$byteChunks): void
+    {
+        $this->randomByteChunks = $byteChunks;
+    }
+
+    private function nextRandomBytes(int $length): string
+    {
+        return array_shift($this->randomByteChunks) ?? random_bytes($length);
+    }
```

PHPMD (2.15.0)

[warning] Avoid unused private methods such as 'nextRandomBytes'. (undefined)

(UnusedPrivateMethod)